src/main/java/com/puppycrawl/tools/checkstyle/Main.java

```diff
@@ -537,18 +546,38 @@ private static int runCheckstyle(CliOptions cliOptions)
                 cliOptions.configLocation, new PropertiesExpander(props),
                 ignoredModulesOptions, multiThreadModeSettings);
 
-        // create a listener for output
-        final AuditListener listener = createListener(cliOptions.format, cliOptions.outputLocation);
-
         // create RootModule object and run it
         final int errorCounter;
         final ClassLoader moduleClassLoader = Checker.class.getClassLoader();
         final RootModule rootModule = getRootModule(config.getName(), moduleClassLoader);
 
         try {
+            final AuditListener listener;
+            if (cliOptions.generateXpathSuppressionsFile) {
+                // create filter to print generated xpath suppressions file
+                final Configuration[] children = config.getChildren();
+                for (Configuration child : children) {
+                    if ("TreeWalker".equals(child.getName())) {
```

Comment:
please make a method `DefaultConfiguration getTreeWalker(Configuration)`
it will make code easy to read

Reply:
done